components/Tabs/tests/unit/Theming/TabsThemingTests.m

```diff
@@ -0,0 +1,183 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import <MaterialComponents/MaterialColorScheme.h>
+#import <MaterialComponents/MaterialTabs.h>
+#import <MaterialComponents/MaterialTypographyScheme.h>
+#import "MaterialContainerScheme.h"
+#import "MaterialTabs+Theming.h"
+
+static const CGFloat kUnselectedOpacity = (CGFloat)0.6;
+
+@interface MDCTabsThemingTest : XCTestCase
+@property(nonatomic, strong) MDCTabBar *tabBar;
+@property(nonatomic, strong) MDCSemanticColorScheme *colorScheme;
+@property(nonatomic, strong) MDCTypographyScheme *typographyScheme;
+@property(nonatomic, strong) MDCContainerScheme *containerScheme;
+@end
+
+@implementation MDCTabsThemingTest
+
+- (void)setUp {
+  [super setUp];
+
+  self.tabBar = [[MDCTabBar alloc] init];
+  self.colorScheme =
+      [[MDCSemanticColorScheme alloc] initWithDefaults:MDCColorSchemeDefaultsMaterial201804];
+  self.typographyScheme =
+      [[MDCTypographyScheme alloc] initWithDefaults:MDCTypographySchemeDefaultsMaterial201804];
+  self.containerScheme = [[MDCContainerScheme alloc] init];
+  self.containerScheme.colorScheme = self.colorScheme;
+  self.containerScheme.typographyScheme = self.typographyScheme;
+}
+
+- (void)tearDown {
+  self.tabBar = nil;
+  self.colorScheme = nil;
+  self.typographyScheme = nil;
+  self.containerScheme = nil;
+
+  [super tearDown];
+}
+
+- (void)testTabBarPrimaryThemingDefault {
+  // When
+  [self.tabBar applyPrimaryThemeWithScheme:self.containerScheme];
+
+  // Then
+  [self verifyTabBarPrimaryTheming];
+}
+
+- (void)testTabBarPrimaryThemingCustom {
+  // Given
+  self.colorScheme = [self customColorScheme];
+  self.typographyScheme = [self customTypographyScheme];
+  self.containerScheme.colorScheme = self.colorScheme;
```

Comment:
Tip: It looks like you can assign `self.containerScheme.colorScheme` directly and you can eliminate the two properties, `colorScheme` and `typographyScheme`.

Reply:
`self.colorScheme` and `self.typographyScheme` are used in other functions, which is the reason why they are instance properties.

Reply:
Yes, but since you could access them via `self.containerScheme` it just seemed like they weren't necessary.

Reply:
Right. I guess it was added for convenience. Talked offline. Will leave it there.